### ISSUE TITLE
Empty helm-pattern when pressing "TAB -> open notmuch-search query buffer"

### ISSUE THREAD

**Issue by Ambrevar** [xuchunyang/helm-notmuch#9]
Working recipe:

- `M-x helm-notmuch`
-  foo
- F2

Failing recipe:

- `M-x helm-notmuch`
-  foo
- TAB
- C-n
- RET

Error:
```
Error: notmuch search requires at least one search term. (see *Notmuch errors* for more details)
```

Indeed, the query is empty, because it's done on `helm-pattern` which is not "foo" once we have pressed TAB.

This is an issue that is general to Helm, but I'm not sure how to tackle it.
@thierryvolpiatto?

**Comment by thierryvolpiatto** [comment]
Pierre Neidhardt <notifications@github.com> writes:

> This is an issue that is general to Helm, but I'm not sure how to tackle it.
> @thierryvolpiatto?

I don't see such error in Helm, TAB is bound to action menu and then the
helm-pattern is reset to nothing until you hit TAB again, when pressing
RET the action under cursor is executed with the candidate selected in
the helm-buffer, what's wrong with this?

For helm-notmutch I don't know, I don't use it.


-- 
Thierry

**Comment by Ambrevar** [comment]
The problem is with the following function:

```
(defun helm-notmuch-search (candidate)
  "Display notmuch query in notmuch-search buffer, highlighting CANDIDATE."
  (notmuch-search (helm-notmuch-maybe-match-incomplete helm-pattern)
                  nil
                  (replace-regexp-in-string "^thread:" "" candidate)))
```

It needs the `helm-pattern` that was used in the prompt outside the
action menu.

So when we are in the action menu, pressing `RET` on this
action will call it with an empty (or different) `helm-pattern`.  What
`helm-notmuch-search` really expect is the `helm-pattern` that was used
before entering the action menu.

Does that make sense?

**Comment by thierryvolpiatto** [comment]
Pierre Neidhardt <notifications@github.com> writes:

> The problem is with the following function:
>
> ```
> (defun helm-notmuch-search (candidate)
> "Display notmuch query in notmuch-search buffer, highlighting CANDIDATE."
> (notmuch-search (helm-notmuch-maybe-match-incomplete helm-pattern)
> nil
> (replace-regexp-in-string "^thread:" "" candidate)))
> ```
>
> It needs the `helm-pattern` that was used in the prompt outside the
> action menu.
>
> So when we are in the action menu, pressing `RET` on this
> action will call it with an empty (or different) `helm-pattern`. What
> `helm-notmuch-search` really expect is the `helm-pattern` that was used
> before entering the action menu.
>
> Does that make sense?

IIUC you want the last value of helm-pattern before switching to helm
action menu?
You can retrieve it in `helm-input-local` if needed.
IOW instead of:

    (notmuch-search (helm-notmuch-maybe-match-incomplete helm-pattern)
use
    (notmuch-search (helm-notmuch-maybe-match-incomplete helm-input-local)

-- 
Thierry

**Comment by Ambrevar** [comment]
I had to use

  (with-helm-buffer helm-input-local)

and it works now!  Thank you Thierry!

@xuchunyang: I'll send another patch for this.
Since you are no longer using it, are you still willing to maintain it?
If not, I'd be happy to adopt it and move it to the Emacs-Helm
organization, if @thierryvolpiatto agrees :)

**Comment by xuchunyang** [comment]
@Ambrevar I don't have much motivation to maintain it. That will be great if you or someone else can maintain it, if @thierryvolpiatto agrees, I will move the repository to the Emacs-Helm organization.

**Comment by thierryvolpiatto** [comment]
Xu Chunyang <notifications@github.com> writes:

> @Ambrevar I don't have much motivation to maintain it. That will be
> great if you or someone else can maintain it, if @thierryvolpiatto
> agrees, I will move the repository to the Emacs-Helm organization.

No problem, you can move it to emacs-helm.

--
Thierry

**Comment by xuchunyang** [comment]
Great, I've moved it to emacs-helm.

@Ambrevar I'm not sure if you have necessary permissions to maintain the repo, such as push code, close issues and merge pull requests. By the way, it seems github redirects the old URL to the new one, so we don't have to update the Melpa recipe.

**Comment by Ambrevar** [comment]
No, I don't have permissions.  Could you grant them to me?  Thanks!

**Comment by xuchunyang** [comment]
It seems that's impossible for me, I didn't see the Settings panel at https://github.com/emacs-helm/helm-notmuch, I think @thierryvolpiatto can grant these permissions to you since he is the owner of the emacs-helm organization.

**Comment by thierryvolpiatto** [comment]
Pierre Neidhardt <notifications@github.com> writes:

> No, I don't have permissions. Could you grant them to me? Thanks!

Done!



-- 
Thierry

**Comment by thierryvolpiatto** [comment]
Xu Chunyang <notifications@github.com> writes:

> Great, I've moved it to emacs-helm.

Thanks.


-- 
Thierry

**Comment by Ambrevar** [comment]
Fixed in d6d552936ff4c6f64c9203bf448310922744627a.

**Comment by Ambrevar** [comment]
Thank you @xuchunyang and @thierryvolpiatto !
